src/main/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProvider.java

```diff
@@ -35,23 +34,25 @@ public static DisplayURLProvider get() {
      *
      * @return all the {@link DisplayURLProvider} implementations.
      */
-    public static Iterable<DisplayURLProvider> all() {
+    public static ExtensionList<DisplayURLProvider> all() {
```

Comment:
This is not breaking since ExtensionList is still Iterable, it just lends itself better to stream API.

Reply:
I'm not sure about this, it causes problems in environments like the PCT: 

```
java.lang.NoSuchMethodError: org.jenkinsci.plugins.displayurlapi.DisplayURLProvider.all()Ljava/lang/Iterable;
	at org.jenkinsci.plugins.blueoceandisplayurl.BlueOceanDisplayURLImplTest.setUp(BlueOceanDisplayURLImplTest.java:178)
```

See [BlueOceanDisplayURLImplTest](https://github.com/jenkinsci/blueocean-display-url-plugin/blob/59bcc13a2c8dff20c6866caa8b2020b039026d57/src/test/java/org/jenkinsci/plugins/blueoceandisplayurl/BlueOceanDisplayURLImplTest.java#L178). That is the only use of `DisplayURLProvider.all()` outside of this plugin that I found from a quick search, and it's in test code, so it doesn't seem to be a big deal. 

Based on the [spec](https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15), I don't think this is a compatible change, since JARs compiled against the old class are looking for a method whose return type is `Iterable`, but not such method exists. Because the new return type is a subtype of the old type, I think the change is _source_ compatible in that all callers can be recompiled against the new version of this plugin without needing to make any changes on their end and things will work correctly, but not _binary_ compatible.